script.js

```diff
@@ -1,4 +1,704 @@
+
+// ********* step 4 : Ace to be come either 1 or 11 ********* //
+//      1) if the total value goes below 21, Ace will be counted as 11
+//      2) if the total value goes above 21, Ace need to change value to 1 instead of 11
+
+//image global variable//
+var image1 =
+  '<img src="https://c.tenor.com/FaMwfY6UU9kAAAAM/shuffle-cards.gif"/>';
+var imageThumbsUp =
+  '<img src="https://c.tenor.com/D5d3KKgydBUAAAAM/gavin-meme.gif"/>';
+var imageWhat =
+  '<img src="https://c.tenor.com/8W03lgoMEv0AAAAM/umm-what.gif"/>';
+var imageEvilLaugh =
+  '<img src="https://c.tenor.com/Jw3KI0JGQFAAAAAM/gavin-gavin-thomas.gif"/>';
+var imageLost =
+  '<img src="https://c.tenor.com/9-0P-Fh_VSMAAAAM/gavin-meme.gif"/>';
+var imageThinking =
+  '<img src="https://c.tenor.com/BMQTJ4nWvCIAAAAM/thinking-think.gif"/>';
+var imageWrongAnswer =
+  '<img src="https://c.tenor.com/S9A24f_TI_EAAAAM/did-you-read-it-you-get-it.gif"/>';
+
+// *** GLOBAL VARIABLE ***//
+var gameMode_Start = "Game Start";
+var gameMode_DrawnCards = "Drawn Cards";
+var gameMode_ResultShown = "Show Result";
+var gameMode_HitOrStand = "Hit or Stand";
+var currentGameMode = gameMode_Start;
+
+// variable for the group of multiple cards holding
+var playerHand = [];
+var dealerHand = [];
+
+// *** FUNCTIONS *** //
+var checkForBlackJack = function (handArray) {
+  // this will return true or false!
+  var playerCardOne = handArray[0];
+  var playerCardTwo = handArray[1];
+  var isBlackJack = false;
+
+  //way to become blackjack (21)
+  //1) ACE + 10 or more  (must have ACE in either one)
+  //2) 10 or more + ACE
+  if (
+    (playerCardOne.name == "ace" && playerCardTwo.rank >= 10) ||
+    (playerCardOne.rank >= 10 && playerCardTwo.name == "ace")
+  ) {
+    isBlackJack = true;
+  }
+  // except that scenario, it would be all NOT black jack and can return directly "false"
+  return isBlackJack;
+};
+
+var calculateTotalHandValue = function (handArray) {
+  var totalHandValue = 0;
+  var aceCounter = 0;
+
+  for (var i = 0; i < handArray.length; i += 1) {
+    var currentCard = handArray[i];
+    // console.log(currentCard);
+    // Jack / Queen / King equals to 10
+    if (
+      currentCard.name == "jack" ||
+      currentCard.name == "queen" ||
+      currentCard.name == "king"
+    ) {
+      totalHandValue = totalHandValue + 10;
+    } else if (currentCard.name == "ace") {
+      totalHandValue = totalHandValue + 11;
+      aceCounter += 1;
+    }
+    // if not, just face value
+    else {
+      totalHandValue = totalHandValue + currentCard.rank;
+    }
+    for (var j = 0; j < aceCounter; j += 1) {
+      if (totalHandValue > 21) {
+        totalHandValue = totalHandValue - 10;
+      }
+    }
+  }
+  return totalHandValue;
+};
+
+var displayPlayerAndDealerHandsFirst = function (
+  playerHandArray,
+  dealerHandArray
+) {
+  //playerHand
+  var playerMessage = "Player Hand : <br>";
+  for (var index = 0; index < playerHandArray.length; index += 1) {
```

Comment:
Take a look at line 116 and 138 as well as 126 and 150. they are very similar, which gives you a hint that each part can probably be a function on its own